lint/location_test.go

```diff
@@ -15,45 +15,113 @@
 package lint
 
 import (
+	"strings"
 	"testing"
+
+	dpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/google/go-cmp/cmp"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/desc/builder"
+	"github.com/jhump/protoreflect/desc/protoparse"
+	"github.com/lithammer/dedent"
 )
 
-func TestLocation_IsValid(t *testing.T) {
-	tests := []struct {
-		l     Location
-		valid bool
-	}{
-		{Location{Position{1, 2}, Position{1, 2}}, true},
-		{Location{Position{0, 0}, Position{1, 1}}, false}, // invalid: start position is invalid
-		{Location{Position{1, 1}, Position{0, 0}}, false}, // invalid: end position is invalid
-		{Location{Position{2, 1}, Position{1, 1}}, false}, // invalid: end line is before start line
-		{Location{Position{1, 2}, Position{1, 1}}, false}, // invalid: end column is before start column in the same line
-		{Location{}, false}, // invalid: empty location
+func TestLocations(t *testing.T) {
+	parser := protoparse.Parser{
+		Accessor: protoparse.FileContentsFromMap(map[string]string{
+			"test.proto": strings.TrimSpace(dedent.Dedent(`
+				// proto3 rules!
+				syntax = "proto3";
+
+				package google.api.linter;
+
+				message Foo {
+				  string bar = 1;
+				}
+			`)),
+		}),
+		IncludeSourceCodeInfo: true,
```

Comment:
Add a test by disabling `IncludeSourceCodeInfo`, and we should not panic, but return `nil` for locations.

Reply:
Ignore it -- it seems you already have such tests.